### ISSUE TITLE
[2018] Dendritic cortical microcircuits approximate the backpropagation algorithm

### ISSUE THREAD

**Issue by rejasupotaro** [rejasupotaro/paper-reading#78]
## Link

https://papers.nips.cc/paper/8089-dendritic-cortical-microcircuits-approximate-the-backpropagation-algorithm

## Note

> The main deep learning mechanism behind these advances (error backpropagation) appears to be at odds with neurobiology.

Machine learning is going through remarkable developments powered by deep neural networks but the workhorse of deep learning is still the classical backpropagation of errors algorithm, which has been long dismissed in neuroscience on the grounds of biologically implausibility. Although neural networks outperform alternative frameworks in accurately reproducing activity patterns observed in the cortex, how the brain could implement backprop-like algorithm remains an open question.

### Why is error-backprop bioimplausible?

- Weight transport problem
  - Suggests that feedback weights = feedforward weights
- Derivative of activation functions
  - Relies on calculating derivatives of the activation functions
- **Two phase learning**
  - (1) Feedforward propagation of activity
  - (2) Error backpropagation
- **Separate error network**
  - Suggest the need for a separate biological network computing the learning rules
- **Non-local learning rules**
  - The weight update depends on non-local information
- Target
  - The target label guides supervised learning

This paper addressed **these** issues.

### Proposal

The prediction errors are encoded at distal dendrites of pyramidal neurons, which receive top-down input from downstream brain areas.

A brain area = A layer in machine learning???

In their model, these errors arise from the inability to exactly match via lateral input from local interneurons the top-down feedback from downstream cortical areas. In contrast to previous approaches, a given neuron is used simultaneously for activity propagation, error encoding, and error propagation to the soma without the need for separate phrases.

### Error-encoding dendritic cortical microcircuits

- Neuron and network model
- Synaptic learning rules
- Comparison to previous work

### Results

- Learning in dendritic error networks approximates backprop
- Deviations from self-predictions encode backpropagated errors
- Network learns to solve a nonlinear regression task
- Microcircuit network learns to classify handwritten digits

## Previous Work

- [1985] A learning algorithm for Boltzmann machines
- [1986] [Learning representations by back-propagating errors](https://www.iro.umontreal.ca/~vincentp/ift3395/lectures/backprop_old.pdf)
- [2011] Spatio-temporal credit assignment in neuronal population learning
- [2014] How auto-encoders could provide credit assignment in deep networks via target propagation
- [2017] Cortical microcircuits as gated-recurrent neural networks

### Back-propagation

Back-propagation is a learning procedure for networks of neuron-like units. The procedure repeatedly adjusts the weights of the connections in the network so as to minimize a measure of the difference between the actual output vector of the net and the desired output vector.

- Perceptron-convergence procedure => Back-propagation
- Adjusted weights of hidden layers represent important features of the task domain
- People tried to find a powerful synaptic modification rule that will allow an arbitrarily connected neural network
- If the input units are directly connected to the output units, it is relatively easy to find learning rules that iteratively adjust the relative strengths of the connections so as to progressively reduce the difference between the actual and desired output vectors

### Credit Assignment Problem

Credit assignment problem (CAP) is the problem of determining the actions that lead to a certain outcome.

How they relate to learning in the brain?

## BibTex

```
@incollection{NIPS2018_8089,
title = {Dendritic cortical microcircuits approximate the backpropagation algorithm},
author = {Sacramento, Jo\~{a}o and Ponte Costa, Rui and Bengio, Yoshua and Senn, Walter},
booktitle = {Advances in Neural Information Processing Systems 31},
editor = {S. Bengio and H. Wallach and H. Larochelle and K. Grauman and N. Cesa-Bianchi and R. Garnett},
pages = {8721--8732},
year = {2018},
publisher = {Curran Associates, Inc.},
url = {http://papers.nips.cc/paper/8089-dendritic-cortical-microcircuits-approximate-the-backpropagation-algorithm.pdf}
}
```
